### PR TITLE
Pr from fork/275

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary 

- This PR pulls the branch from a fork into the RAJAPerf repo so all CI checks can run.
- The original PR is https://github.com/LLNL/RAJAPerf/pull/275
- Please approve this PR, not the original as this one will be merged.

### CODE REVIEW

**Review by artv3 — APPROVED** [review]

File: src/algorithm/MEMCPY-OMPTarget.cpp [hunk 1]
```diff
@@ -90,4 +92,4 @@ void MEMCPY::runOpenMPTargetVariant(VariantID vid, size_t RAJAPERF_UNUSED_ARG(tu
 } // end namespace algorithm
 } // end namespace rajaperf
 
-#endif  // RAJA_ENABLE_TARGET_OPENMP
+#endif  // RAJA_ENABLE_TARGET_OPENMP
```
Comment (MrBurmark) [inline src/algorithm/MEMCPY-OMPTarget.cpp:95]: What about these no newline at end of file formatting changes?

Comment (rhornung67) [reply src/algorithm/MEMCPY-OMPTarget.cpp:95]: @MrBurmark do you want to push changes to fix those. Not sure if @ThorBl has permissions to do that.

Comment (MrBurmark) [inline src/algorithm/MEMCPY-OMPTarget.cpp:95]:
```suggestion
#endif  // RAJA_ENABLE_TARGET_OPENMP

```

File: src/basic/REDUCE_STRUCT-OMPTarget.cpp [hunk 2]
```diff
@@ -123,4 +123,4 @@ void REDUCE_STRUCT::runOpenMPTargetVariant(VariantID vid)
 } // end namespace basic
 } // end namespace rajaperf
 
-#endif  // RAJA_ENABLE_TARGET_OPENMP
+#endif  // RAJA_ENABLE_TARGET_OPENMP
```
Comment (MrBurmark) [inline src/basic/REDUCE_STRUCT-OMPTarget.cpp:126]:
```suggestion
#endif  // RAJA_ENABLE_TARGET_OPENMP

```

File: src/algorithm/MEMCPY-OMPTarget.cpp [hunk 3]
```diff
@@ -30,10 +30,12 @@ namespace algorithm
   int hid = omp_get_initial_device(); \
   int did = omp_get_default_device(); \
 \
-  allocAndInitOpenMPDeviceData(x, m_x, iend, did, hid);
+  allocAndInitOpenMPDeviceData(x, m_x, iend, did, hid); \
+  allocAndInitOpenMPDeviceData(y, m_y, iend, did, hid);
 
 #define MEMCPY_DATA_TEARDOWN_OMP_TARGET \
   deallocOpenMPDeviceData(x, did); \
+  deallocOpenMPDeviceData(y, did);
```
Comment (MrBurmark) [inline src/algorithm/MEMCPY-OMPTarget.cpp:38]: Looking at this makes me think we probably need to copy y back to m_y.

Comment (MrBurmark) [reply src/algorithm/MEMCPY-OMPTarget.cpp:38]:
```suggestion
  getOpenMPDeviceData(m_y, y, iend, hid, did); \
  deallocOpenMPDeviceData(x, did); \
  deallocOpenMPDeviceData(y, did);
```

Comment (rhornung67) [reply src/algorithm/MEMCPY-OMPTarget.cpp:38]: We also don't need the "\" line continuation characters here.

Comment (MrBurmark) [reply src/algorithm/MEMCPY-OMPTarget.cpp:39]: I'm not sure what you mean.

**Comment by MrBurmark** [comment]
I discovered some problems in RAJA when attempting to compile this, looks like we'll need more hotfixes.

File: src/algorithm/MEMCPY-OMPTarget.cpp (hunk 3 above)
Comment (rhornung67) [reply src/algorithm/MEMCPY-OMPTarget.cpp:39]: the backslash. sorry it didn't appear in my comment.

Comment (rhornung67) [reply src/algorithm/MEMCPY-OMPTarget.cpp:39]: nevermind. it's part of the macro. so it's needed. Please disregard my comments.

**Comment by rhornung67** [comment]
> I discovered some problems in RAJA when attempting to compile this, looks like we'll need more hotfixes.

What were the problems? Was this code ever run and tested?

**Comment by MrBurmark** [comment]
It looks like this particular path wasn't tested. I also got a lot of signed unsigned comparison warnings, but as far as I can tell everything was the same type so that may be a clang issue.

**Comment by rhornung67** [comment]
@MrBurmark we should get this one merged too. Please take a look when you have a minute.

**Comment by artv3** [comment]
> 

Can you post where those are coming from?

**Comment by artv3** [comment]
> It looks like the path mentioned in the PR above wasn't tested. I also got a lot of signed unsigned comparison warnings, but as far as I can tell everything was the same type so that may be a clang issue.

Can you post where those are coming from?

**Comment by rhornung67** [comment]
@MrBurmark I merged in develop and resolved conflicts. I think this will be ready to merge when it comes back green. Please review when you have time.

**Comment by rhornung67** [comment]
@artv3 I discussed the signed/unsigned comparison warnings with @MrBurmark a couple of days ago. They don't appear with XL, for example. We think it's a compiler issue.

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**